core/services/ocr2/plugins/ocr2keeper/evm21/transmit/encoding.go

```diff
@@ -0,0 +1,111 @@
+package transmit
+
+import (
+	"fmt"
+	"math/big"
+
+	ocr2keepers "github.com/smartcontractkit/ocr2keepers/pkg/v3/types"
+
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	iregistry21 "github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/i_keeper_registry_master_wrapper_2_1"
+)
+
+func defaultLogParser(registry *iregistry21.IKeeperRegistryMaster, log logpoller.Log) (transmitEventLog, error) {
```

Comment:
Should defaultLogParser be a type with an internal `registry *iregistry21.IKeeperRegistryMaster` type, and a `Parse` function that accepts a `logpoller.Log`?

Reply:
We can also wrap it with a struct, I used a function for simplicity